src/overlays/actors/ovl_En_Part/z_en_part.c

```diff
@@ -28,16 +30,114 @@ const ActorInit En_Part_InitVars = {
     (ActorFunc)EnPart_Draw,
 };
 
-#endif
+void EnPart_Init(Actor* thisx, PlayState* play) {
+}
+
+void EnPart_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void func_80865390(EnPart* this, PlayState* play) {
+    f32 randVar;
+
+    this->unk144 = 1;
+    this->actor.world.rot.y = (Rand_ZeroOne() * 20000.0f);
+    switch (this->actor.params) {
+        case PARAMS_1:
+        case PARAMS_4:
+            randVar = Rand_ZeroOne() * 17.0f;
+            this->unk146 += (s16)randVar + 5;
+            this->actor.velocity.y = (Rand_ZeroOne() * 5.0f) + 4.0f;
+            this->actor.gravity = (-0.6f) - (Rand_ZeroOne() * 0.5f);
+            this->unk14C = 0.15f;
+            break;
+        case PARAMS_15:
+            this->actor.world.rot.y = this->actor.parent->shape.rot.y + 0x8000;
+            this->unk146 = 100;
+            this->actor.velocity.y = 7.0f;
+            this->actor.speedXZ = 2.0f;
+            this->actor.gravity = -1.0f;
+            break;
+    }
+}
+
+void func_808654C4(EnPart* this, PlayState* play) {
+    s16 effectScale;
+    Vec3f effectPos;
+    Vec3f effectVelocity = { 0.0f, 0.0f, 0.0f };
+    s32 i;
+    Vec3f pos;
+
+    if (this->actor.params == PARAMS_15) {
+        this->unk146--;
+        if (this->unk146 > 0) {
+            this->actor.shape.rot.x += 0x3a98;
+            this->actor.shape.rot.y = this->actor.shape.rot.y;
+            this->actor.shape.rot.z = this->actor.shape.rot.z;
+            if (BgCheck_SphVsFirstPoly(&play->colCtx, &this->actor.world.pos, 20.0f)) {
+                this->unk146 = 0;
+            }
+        } else {
+            Math_Vec3f_Copy(&pos, &this->actor.world.pos);
+            pos.y = this->actor.floorHeight;
+            func_800B3030(play, &pos, &gZeroVec3f, &gZeroVec3f, this->actor.scale.y * 1400.0f, 7, 0);
+            SoundSource_PlaySfxAtFixedWorldPos(play, &this->actor.world.pos, 10, NA_SE_EN_EXTINCT);
+            Actor_MarkForDeath(&this->actor);
+        }
+    } else {
+        if (this->unk146 <= 0) {
```

Comment:
```suggestion
    } else if (this->unk146 <= 0) {
```
Can these two be merged? Would need to remove the closing bracket and make the final else match this indent. If it doesn't match don't worry about it.